src/main.rs

```diff
@@ -543,6 +543,28 @@ fn cmd_inspect(
         version: version.clone(),
     });
 
+    if sub_args.mode == FetchMode::Sourcegraph {
+        tokio::runtime::Handle::current()
+            .block_on(prompt_criteria_eulas(
+                out,
+                cfg,
+                network.as_ref(),
+                &store,
+                package,
+                None,
+                version,
+            ))
+            .into_diagnostic()?;
+
+        let url = format!("https://sourcegraph.com/crates/{package}@v{version}");
+        match open::that(&url) {
+            Ok(()) => return Ok(()),
+            Err(e) => {
+                warn!("Couldn't open browser, falling back to local {:?}", e);
```

Comment:
This fallback behaviour is kinda terrible, given that if we hit it we'll prompt the user to accept EULAs again... It would be nice if we could hold ourselves to only trying to do that once. Perhaps we could define a local closure like:

```rust
let mut did_prompt_eulas = false;
let mut prompt_eulas = || async {
  if did_prompt_eulas {
    return Ok(());
  }
  did_prompt_eulas = true;
  return prompt_criteria_eulas(out, cfg, network.as_ref(), &store, package, None, version).await;
};
```